src/menu/actions.c

```diff
@@ -81,7 +94,16 @@ static void actions_update_direction (menu_t *menu) {
 }
 
 static void actions_update_buttons (menu_t *menu) {    
-    joypad_buttons_t pressed = joypad_get_buttons_pressed(JOYPAD_PORT_1);
+    int prev_pad = -1;
+    joypad_buttons_t pressed = {0};
+
+    JOYPAD_PORT_FOREACH (i) {
+        joypad_buttons_t pressed_temp = joypad_get_buttons_pressed(i);
```

Comment:
I don't think the temp variable is even necessary here. Just assign the `pressed` variable and exit the loop when currently checked controller has any pressed button(s).